boto/dynamodb2/layer1.py

```diff
@@ -169,7 +169,7 @@ def __init__(self, **kwargs):
                     break
 
         # Only set host if it isn't manually overwritten
-        if 'host' not in kwargs:
+        if kwargs.get('host') is None:
             kwargs['host'] = region.endpoint
```

Comment:
`kwargs['host'] = kwargs.get('host') or region.endpoint` # so at least it's consistently defined – and if someone passes 0 or False, it will behave the same.